### DISCUSSION TITLE
How to change colors in PlotFootprint?

### DISCUSSION THREAD

**Discussion by ericli0419** [stuart-lab/signac#979] [Q&A] [answered]
Hi,
Is there a way to change the colors in PlotFootprint?
Thanks so much!

**Comment by wudustan** [comment, marked as answer]
PlotFootprint() generates a ggplot object so you can use typical ggplot options:

    cluster.cols # a vector of colours for each cluster
    names(cluster.cols) <- signac.object$seurat_clusters

    PlotFootprint(signac.object, ..., group.by = "seurat_clusters") +
    scale_colour_manual(values = cluster.cols)

**Reply by ericli0419** [reply]
Hi,
thanks for your response! I used your code but the color still doesn't change. I have tried other ggplot options, also can't change the color.

**Reply by wudustan** [reply]
Can you post an example of your code? Note, if you are using patchwork, then the proper way to change the colour of a set of patched gg objects is to use `&` instead of `+`

**Reply by ericli0419** [reply]
I have used the following code.

```
CelltypeColors=c('Microglia1'='#F47D2B','Microglia2'='#FEE500')

PlotFootprint(sce, features = "FOSL2::JUN",group.by = "celltype") +
scale_colour_manual(values = CelltypeColors)

PlotFootprint(sce, features = "FOSL2::JUN",group.by = "celltype")
&scale_fill_manual(values = CelltypeColors)

PlotFootprint(sce, features = "FOSL2::JUN",group.by = "celltype")
&scale_fill_manual(values = unname(CelltypeColors))
```

**Reply by wudustan** [reply]
PlotFootprint(sce, features = "FOSL2::JUN",group.by = "celltype") +
    scale_colour_manual(values = CelltypeColors)

This is very similar to the code I use and it colours the lines with no issues. I guess I would suggest checking your `sce$celltype` unique elements match `Microglia1` / `Microglia2` with no spaces in the string etc or that they aren't being inadvertently coerced to a factor numeric or something like that.

**Reply by ericli0419** [reply]
> ```
> PlotFootprint(sce, features = "FOSL2::JUN",group.by = "celltype") +
> scale_colour_manual(values = CelltypeColors)
> ```
> 
> This is very similar to the code I use and it colours the lines with no issues. I guess I would suggest checking your `sce$celltype` unique elements match `Microglia1` / `Microglia2` with no spaces in the string etc or that they aren't being inadvertently coerced to a factor numeric or something like that.

The Signac object and CelltypeColors worked well elsewhere, and I will check again. Thank you very much for your advice and reply!
